src/PowerShellEditorServices/Services/Analysis/AnalysisService.cs

```diff
@@ -51,711 +78,279 @@ public class AnalysisService : IDisposable
+        /// Get comment help text for a PowerShell function definition.
         /// </summary>
-        /// <param name="scriptDefinition">Script text to be formatted</param>
-        /// <param name="settings">ScriptAnalyzer settings</param>
-        /// <param name="rangeList">The range within which formatting should be applied.</param>
-        /// <returns>The formatted script text.</returns>
-        public async Task<string> FormatAsync(
-            string scriptDefinition,
-            Hashtable settings,
-            int[] rangeList)
+        /// <param name="functionText">The text of the function to get comment help for.</param>
+        /// <param name="helpLocation">A string referring to which location comment help should be placed around the function.</param>
+        /// <param name="forBlockComment">If true, block comment help will be supplied.</param>
+        /// <returns></returns>
+        public async Task<string> GetCommentHelpText(string functionText, string helpLocation, bool forBlockComment)
         {
-            // We cannot use Range type therefore this workaround of using -1 default value.
-            // Invoke-Formatter throws a ParameterBinderValidationException if the ScriptDefinition is an empty string.
-            if (string.IsNullOrEmpty(scriptDefinition))
+            if (AnalysisEngine == null)
             {
                 return null;
             }
 
-            var argsDict = new Dictionary<string, object> {
-                    {"ScriptDefinition", scriptDefinition},
-                    {"Settings", settings}
-            };
-            if (rangeList != null)
-            {
-                argsDict.Add("Range", rangeList);
-            }
+            Hashtable commentHelpSettings = AnalysisService.GetCommentHelpRuleSettings(helpLocation, forBlockComment);
 
-            PowerShellResult result = await InvokePowerShellAsync("Invoke-Formatter", argsDict).ConfigureAwait(false);
+            ScriptFileMarker[] analysisResults = await AnalysisEngine.AnalyzeScriptAsync(functionText, commentHelpSettings).ConfigureAwait(false);
 
-            if (result == null)
+            if (analysisResults.Length == 0
+                || analysisResults[0]?.Correction?.Edits == null
+                || analysisResults[0].Correction.Edits.Count() == 0)
             {
-                _logger.LogError("Formatter returned null result");
                 return null;
             }
 
-            if (result.HasErrors)
-            {
-                var errorBuilder = new StringBuilder().Append(s_indentJoin);
-                foreach (ErrorRecord err in result.Errors)
-                {
-                    errorBuilder.Append(err).Append(s_indentJoin);
-                }
-                _logger.LogWarning($"Errors found while formatting file: {errorBuilder}");
-                return null;
-            }
+            return analysisResults[0].Correction.Edits[0].Text;
```

Comment:
```suggestion
            return analysisResults[0]?.Correction?.Edits[0]?.Text;
```
why not something like this instead of the if?

Reply:
That doesn't defend against the possibility of having empty collections